x-pack/solutions/observability/plugins/streams/server/lib/streams/client.ts

```diff
@@ -444,6 +453,29 @@ export class StreamsClient {
     return { parentDefinition };
   }
 
+  /**
+   * Validates the members of the group streams to ensure they are NOT
+   * GroupStreamDefinitions
+   */
+  async validateGroupStream({ definition }: { definition: GroupStreamDefinition }) {
+    const { members } = definition.group;
+
+    if (members.includes(definition.name)) {
+      throw new ForbiddenMemberTypeError('The same stream can not be part of the group');
```

Comment:
```suggestion
      throw new ForbiddenMemberTypeError('Group streams can not include themselves as a member');
```

Reply:
But your original error is also an interesting thing to check, if there are duplicate members